### DISCUSSION TITLE
when creating a new ChatRequest, where do I define the assistant?

### DISCUSSION THREAD

**Discussion by marcusx2** [RageAgainstThePixel/com.openai.unity#222] [Q&A] [answered]
For example, I get the assistant

`assistant = await client.AssistantsEndpoint.RetrieveAssistantAsync(assistanId);`

but how do I use it with the chat request?

`var request = new ChatRequest(conversation.Messages, tools: assistantTools);`

**Comment by StephenHodgson** [comment, marked as answer]
I think you've got the two features confused. Assistants is a completely different thing from Chats.

Chats is a predecessor of the more advanced Assistant features OpenAI has made and cannot be mixed and matched. 

Check the [Assistants](https://github.com/RageAgainstThePixel/com.openai.unity#assistants) and [Threads](https://github.com/RageAgainstThePixel/com.openai.unity#threads) docs for detailed examples

**Comment by marcusx2** [comment]
Ok thank you for the fast response. Here's how I'm currently doing it

```
assistant = await client.AssistantsEndpoint.RetrieveAssistantAsync(assistanId);
thread = await client.ThreadsEndpoint.CreateThreadAsync();
var run = await thread.CreateRunAsync(assistant);
run = await run.WaitForStatusChangeAsync();
var responses = await run.ListMessagesAsync();
string result = responses.Items.First().PrintContent();
```

The result, openAI's response, doesn't match what I get when playing with the playground on the dashboard. So I tried using Chats because there's an example scene using that. Is there something I'm doing wrong on the code above, or am I missing something?

**Comment by marcusx2** [comment]
Or rather maybe can you send me a modified version of ChatBehavior.cs that uses Assistants and Threads? That'd be incredibly helpful. My email is marjob14@gmail.com . Thank you so much for your quick replies.

**Reply by StephenHodgson** [reply]
There currently isn't a sample behavior that implements them. See https://github.com/RageAgainstThePixel/com.openai.unity/issues/162
